src/bin/ion/commands/beta/schema/validate.rs

```diff
@@ -0,0 +1,142 @@
+use ion_schema::external::ion_rs::text::writer::TextWriter;
+use ion_schema::external::ion_rs::IonType;
+use ion_schema::external::ion_rs::value::writer::{TextKind, Format, ElementWriter};
+use std::str::from_utf8;
+
+const ABOUT: &str = "validates an Ion Value based on given Ion Schema Type";
+
+// Creates a `clap` (Command Line Arguments Parser) configuration for the `load` command.
+// This function is invoked by the `load` command's parent `schema`, so it can describe its
+// child commands.
+pub fn app() -> App<'static, 'static> {
+    App::new("validate")
+        .about(ABOUT)
+        .arg(
+            // Schema file can be specified by the "-s" or "--schema" flags.
+            Arg::with_name("schema")
+                .long("schema")
+                .short("s")
+                .required(true)
+                .takes_value(true)
+                .value_name("SCHEMA")
+                .help("The Ion Schema file to load"),
+        )
+        .arg(
+            // Directory(s) that will be used as authority(s) for schema system
+            Arg::with_name("directories")
+                .long("directory")
+                .short("d")
+                .min_values(1)
+                .takes_value(true)
+                .multiple(true)
+                .value_name("DIRECTORY")
+                .required(true)
+                .help("One or more directories that will be searched for the requested schema"),
+        )
+        .arg(
+            // Input ion file can be specified by the "-i" or "--input" flags.
+            Arg::with_name("input")
+                .long("input")
+                .short("i")
+                .required(true)
+                .takes_value(true)
+                .value_name("INPUT_FILE")
+                .help("Input file containing the Ion values to be validated"),
+        )
+        .arg(
+            // Schema Type can be specified by the "-t" or "--type" flags.
+            Arg::with_name("type")
+                .long("type")
+                .short("t")
+                .required(true)
+                .takes_value(true)
+                .value_name("TYPE")
+                .help("Name of schema type from given schema that needs to be used for validation"),
+        )
+}
+
+// This function is invoked by the `load` command's parent `schema`.
+pub fn run(_command_name: &str, matches: &ArgMatches<'static>) -> Result<()> {
+    // Extract the user provided document authorities/ directories
+    let authorities: Vec<_> = matches.values_of("directories").unwrap().collect();
+
+    // Extract schema file provided by user
+    let schema_id = matches.value_of("schema").unwrap();
+
+    // Extract the schema type provided by user
+    let schema_type = matches.value_of("type").unwrap();
+
+    // Extract Ion value provided by user
+    let input_file = matches.value_of("input").unwrap();
+    let value = fs::read(input_file).expect("Can not load given ion file");
```

Comment:
I would expect bad input files to be a common occurrence. I think we should return a human-friendly `anyhow` error if this happens. Here's [an example from `ion beta inspect`](https://github.com/amzn/ion-cli/blob/a1378f7f8b3757ebf98c1acc2d68db76df1927c8/src/bin/ion/commands/beta/inspect.rs#L110).